assets/js/utils/products.js

```diff
@@ -1,25 +1,29 @@
 /**
  * Get the src of the first image attached to a product (the featured image).
  *
- * @param {array} images The array of images, destructured from the product object.
+ * @param {object} product The product object to get the images from.
+ * @param {array} product.images The array of images, destructured from the product object.
  * @return {string} The full URL to the image.
  */
-export function getImageSrcFromProduct( { images = [] } ) {
-	if ( images.length ) {
-		return images[ 0 ].src || '';
+export function getImageSrcFromProduct( product ) {
+	if ( ! product || ! product.images || ! product.images.length ) {
```

Comment:
I like this (and the similar refactor for `getImageIdFromProduct`) too 👍 It better aligns with the name of the function.